testing/async/exclusive_scan/basic.cu

```diff
@@ -0,0 +1,72 @@
+#include <thrust/detail/config.h>
+
+#if THRUST_CPP_DIALECT >= 2014
+
+#include <async/test_policy_overloads.h>
+
+#include <async/exclusive_scan/mixin.h>
+
+template <typename input_value_type,
+          typename output_value_type   = input_value_type,
+          typename initial_value_type  = input_value_type,
+          typename alternate_binary_op = thrust::maximum<>>
+struct basic_invoker
+    : testing::async::mixin::input::device_vector<input_value_type>
+    , testing::async::mixin::output::device_vector<output_value_type>
+    , testing::async::exclusive_scan::mixin::postfix_args::
+        all_overloads<initial_value_type, alternate_binary_op>
+    , testing::async::exclusive_scan::mixin::invoke_reference::
+        host_synchronous<input_value_type, output_value_type>
+    , testing::async::exclusive_scan::mixin::invoke_async::basic
+    , testing::async::mixin::compare_outputs::assert_maybe_fuzzy_equal_quiet
+{
+  static std::string description()
+  {
+    return "basic invocation with device vectors";
+  }
+};
+
+template <typename T>
+struct TestBasic
```

Comment:
We really ought to unify naming conventions to `snake_case`; I prefer that new stuff at least use `snake_case`

Reply:
I'd prefer to have `exclusive_scan` in the name to be consistent with other tests.

Reply:
And perhaps call this "simple", because I think that's the convention of the other tests (instead of "basic").

Reply:
1. Changed to `snake_case` for the all new async test ✔
2. `exclusive_scan` appears in the executable and test name. These are subtests of `exclusive_scan`, including the algorithm name for each subtest here would be redundant and verbose.
3. Changed `basic` to `simple`.